src/main/java/org/terasology/metalrenegades/ai/system/NeedsSystem.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.ai.system;
+
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.metalrenegades.ai.CitizenNeed;
+import org.terasology.metalrenegades.ai.component.NeedsComponent;
+import org.terasology.registry.In;
+
+/**
+ * Manages needs for all citizens with {@link NeedsComponent}.
+ */
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class NeedsSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
```

Comment:
I think it would be better to align with the world time. How about converting this into a subscriber to `WorldTimeEvent`? It is triggered 100 times per world day.

Reply:
I changed this to the world time event, however this takes approx. half the time per cycle as the old timer, so I also halved all the need reduction rates 👍